connect/src/com/telenor/connect/AbstractSdkProfile.java

```diff
@@ -166,6 +166,10 @@ protected void previewParameters(ParametersHolder parameters) {
                         && isEmpty(wellKnownConfig.getNetworkAuthenticationTargetUrls()))) {
                     parameters.add("login_hint", String.format("MCCMNC:%s", mccMnc));
         }
+
+        if (!ConnectSdk.isCellularDataNetworkConnected()) {
```

Comment:
But then again I think you might have problems with isCellularDataNetworkConnected.
If both wifi and cellular are on, this method will return false. That happens by design AFAIK, or at least happened with the device I tested with it. I recall now, this is why I took the callback approach, as it's the only one that worked for me.
If you can't fix isCellularDataNetworkConnected to behave sane, I'm afraid the hotspot could also not be solve in a similar way.